wallet/src/scheme/freeutxo.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{
+    scheme::{on_tx_input, on_tx_output},
+    states::{States, Status},
+    store::UtxoStore,
+};
+use chain_crypto::{Ed25519, Ed25519Extended, PublicKey, SecretKey};
+use chain_impl_mockchain::{
+    fragment::{Fragment, FragmentId},
+    transaction::{InputEnum, UtxoPointer},
+    value::Value,
+};
+use chain_path_derivation::{AnyScheme, DerivationPath};
+use ed25519_bip32::DerivationScheme;
+use hdkeygen::Key;
+
+pub struct Wallet {
+    state: States<FragmentId, UtxoStore<SecretKey<Ed25519Extended>, AnyScheme>>,
```

Comment:
using `AnyScheme` here is problematic, because then the UtxoStore groups all utxos in the same path (and that's the reason the test is failing). @NicolasDP is this by design? The problem is that when I add two different utxo's with different keys, they get grouped together when I try to recover the private key for signing.